src/lib/server/endpoints/aws/endpointBedrock.ts

```diff
@@ -0,0 +1,150 @@
+	BedrockRuntimeClient,
+	InvokeModelWithResponseStreamCommand,
+} from "@aws-sdk/client-bedrock-runtime";
+import { createImageProcessorOptionsValidator, makeImageProcessor } from "../images";
+import type { EndpointMessage } from "../endpoints";
+import type { MessageFile } from "$lib/types/Message";
+
+export const endpointBedrockParametersSchema = z.object({
+	weight: z.number().int().positive().default(1),
+	type: z.literal("bedrock"),
+	region: z.string().default("us-east-1"),
+	model: z.any(),
+	anthropicVersion: z.string().default("bedrock-2023-05-31"),
+	multimodal: z
+		.object({
+			image: createImageProcessorOptionsValidator({
+				supportedMimeTypes: [
+					"image/png",
+					"image/jpeg",
+					"image/webp",
+					"image/avif",
+					"image/tiff",
+					"image/gif",
+				],
+				preferredMimeType: "image/webp",
+				maxSizeInMB: Infinity,
+				maxWidth: 4096,
+				maxHeight: 4096,
+			}),
+		})
+		.default({}),
+});
+
+export async function endpointBedrock(
+	input: z.input<typeof endpointBedrockParametersSchema>
+): Promise<Endpoint> {
+	const { region, model, anthropicVersion, multimodal } =
+		endpointBedrockParametersSchema.parse(input);
+	const client = new BedrockRuntimeClient({
+		region,
+	});
+	const imageProcessor = makeImageProcessor(multimodal.image);
+
+	return async ({ messages, preprompt, generateSettings }) => {
+		let system = preprompt;
+		// Use the first message as the system prompt if it's of type "system"
+		if (messages?.[0]?.from === "system") {
+			system = messages[0].content;
+			messages = messages.slice(1); // Remove the first system message from the array
+		}
+
+		const formattedMessages = await prepareMessages(messages, imageProcessor);
+
+		let tokenId = 0;
+		const parameters = { ...model.parameters, ...generateSettings };
+		return (async function* () {
+			const command = new InvokeModelWithResponseStreamCommand({
+				body: Buffer.from(
+					JSON.stringify({
+						anthropic_version: anthropicVersion,
+						max_tokens: parameters.max_new_tokens ? parameters.max_new_tokens : 4096,
+						messages: formattedMessages,
+						system,
+					}),
+					"utf-8"
+				),
+				contentType: "application/json",
+				accept: "application/json",
+				modelId: model.id,
+				trace: "DISABLED",
+			});
+
+			const response = await client.send(command);
+
+			let text = "";
+
+			for await (const item of response.body ?? []) {
+				const chunk = JSON.parse(new TextDecoder().decode(item.chunk?.bytes));
+				const chunk_type = chunk.type;
+
+				if (chunk_type === "content_block_delta") {
+					text += chunk.delta.text;
+					yield {
+						token: {
+							id: tokenId++,
+							text: chunk.delta.text,
+							logprob: 0,
+							special: false,
+						},
+						generated_text: null,
+						details: null,
+					} satisfies TextGenerationStreamOutput;
+				} else if (chunk_type === "message_stop") {
+					yield {
+						token: {
+							id: tokenId++,
+							text: "",
+							logprob: 0,
+							special: false,
```

Comment:
I think you can get around your issue above by setting `special: true` in the last message

Reply:
Yes, that fixed it. Thank you!